src/client.rs

```diff
@@ -109,15 +102,8 @@ pub async fn client_main(
             let mut message = thread_receiver.try_recv();
             while message.is_ok() {
                 match message.unwrap() {
-                    // Sync our state to the parent.
-                    GooseClientCommand::SYNC => {
-                        thread_sender.send(thread_client.clone()).unwrap();
-                        // Reset per-thread counters, as totals have been sent to the parent
-                        thread_client.requests = HashMap::new();
-                    }
-                    // Sync our state to the parent and then exit.
+                    // Time to exit.
```

Comment:
This reduces some complexity: we no longer need the parent to ask the client threads to send statistics, as now the client thread just sends statistics in real-time.